File: frontend/src/types/Event.ts
export interface Event {
  id: string;
  title: string;
  description?: string;
  startTime: string;
  endTime?: string;
  location?: string;
  
  // Creator details
  creatorId: string;
  creatorName: string;
  creatorProfilePicUrl?: string;
  
  // Group details (if associated)
  groupId?: string;
  groupName?: string;
  
  category: EventCategory;
  status: EventStatus;
  maxAttendees?: number;
  isRecurring: boolean;
  recurrenceType?: RecurrenceType;
  recurrenceEndDate?: string;
  requiresApproval: boolean;
  
  // RSVP summary
  rsvpSummary?: EventRsvpSummary;
  
  createdAt: string;
  updatedAt: string;
}

export interface EventRequest {
  title: string;
  description?: string;
  startTime: string;
  endTime?: string;
  location?: string;
  groupId?: string;
  category?: EventCategory;
  status?: EventStatus;
  maxAttendees?: number;
  isRecurring?: boolean;
  recurrenceType?: RecurrenceType;
  recurrenceEndDate?: string;
  requiresApproval?: boolean;
}

export interface EventRsvp {
  eventId: string;
  eventTitle: string;
  eventStartTime: string;
  eventLocation?: string;
  userId: string;
  userName: string;
  userProfilePicUrl?: string;
  response: RsvpResponse;
  guestCount: number;
  notes?: string;
  timestamp: string;
  updatedAt: string;
}

export interface EventRsvpRequest {
  eventId: string;
  response: RsvpResponse;
  guestCount?: number;
  notes?: string;
}

export interface EventRsvpSummary {
  yesCount: number;
  noCount: number;
  maybeCount: number;
  totalResponses: number;
  totalAttendees: number;
  
  // User's own RSVP status
  userRsvpResponse?: string;
  userGuestCount?: number;
  userNotes?: string;
}

export interface EventsResponse {
  events: Event[];
  totalPages: number;
  totalElements: number;
  currentPage: number;
  pageSize: number;
}

export enum EventCategory {
  GENERAL = 'GENERAL',
  WORSHIP = 'WORSHIP',
  BIBLE_STUDY = 'BIBLE_STUDY',
  PRAYER = 'PRAYER',
  FELLOWSHIP = 'FELLOWSHIP',
  OUTREACH = 'OUTREACH',
  YOUTH = 'YOUTH',
  CHILDREN = 'CHILDREN',
  MENS = 'MENS',
  WOMENS = 'WOMENS',
  SENIORS = 'SENIORS',
  MISSIONS = 'MISSIONS',
  MINISTRY = 'MINISTRY',
  SOCIAL = 'SOCIAL',
  EDUCATION = 'EDUCATION',
  MUSIC = 'MUSIC',
  OTHER = 'OTHER'
}

export enum EventStatus {
  SCHEDULED = 'SCHEDULED',
  CANCELLED = 'CANCELLED',
  COMPLETED = 'COMPLETED',
  POSTPONED = 'POSTPONED'
}

export enum RecurrenceType {
  DAILY = 'DAILY',
  WEEKLY = 'WEEKLY',
  MONTHLY = 'MONTHLY',
  YEARLY = 'YEARLY'
}

export enum RsvpResponse {
  YES = 'YES',
  NO = 'NO',
  MAYBE = 'MAYBE'
}

// Helper functions for display
export const getEventCategoryDisplay = (category: EventCategory): string => {
  const displays: Record<EventCategory, string> = {
    [EventCategory.GENERAL]: 'General',
    [EventCategory.WORSHIP]: 'Worship Service',
    [EventCategory.BIBLE_STUDY]: 'Bible Study',
    [EventCategory.PRAYER]: 'Prayer Meeting',
    [EventCategory.FELLOWSHIP]: 'Fellowship',
    [EventCategory.OUTREACH]: 'Outreach',
    [EventCategory.YOUTH]: 'Youth Ministry',
    [EventCategory.CHILDREN]: 'Children Ministry',
    [EventCategory.MENS]: "Men's Ministry",
    [EventCategory.WOMENS]: "Women's Ministry",
    [EventCategory.SENIORS]: 'Senior Ministry',
    [EventCategory.MISSIONS]: 'Missions',
    [EventCategory.MINISTRY]: 'Ministry',
    [EventCategory.SOCIAL]: 'Social Event',
    [EventCategory.EDUCATION]: 'Education',
    [EventCategory.MUSIC]: 'Music Ministry',
    [EventCategory.OTHER]: 'Other'
  };
  return displays[category] || category;
};

export const getEventStatusDisplay = (status: EventStatus): string => {
  const displays: Record<EventStatus, string> = {
    [EventStatus.SCHEDULED]: 'Scheduled',
    [EventStatus.CANCELLED]: 'Cancelled',
    [EventStatus.COMPLETED]: 'Completed',
    [EventStatus.POSTPONED]: 'Postponed'
  };
  return displays[status] || status;
};

export const getRsvpResponseDisplay = (response: RsvpResponse): string => {
  const displays: Record<RsvpResponse, string> = {
    [RsvpResponse.YES]: 'Attending',
    [RsvpResponse.NO]: 'Not Attending',
    [RsvpResponse.MAYBE]: 'Maybe'
  };
  return displays[response] || response;
};

File: backend/src/main/java/com/churchapp/entity/Event.java
package com.churchapp.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "events", indexes = {
    @Index(name = "idx_event_creator_id", columnList = "creator_id"),
    @Index(name = "idx_event_group_id", columnList = "group_id"),
    @Index(name = "idx_event_start_time", columnList = "start_time"),
    @Index(name = "idx_event_end_time", columnList = "end_time"),
    @Index(name = "idx_event_created_at", columnList = "created_at")
})
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Event {
    
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id", updatable = false, nullable = false)
    private UUID id;
    
    @NotBlank(message = "Event title is required")
    @Size(min = 3, max = 200, message = "Event title must be between 3 and 200 characters")
    @Column(name = "title", nullable = false, length = 200)
    private String title;
    
    @Size(max = 2000, message = "Event description cannot exceed 2000 characters")
    @Column(name = "description", columnDefinition = "TEXT")
    private String description;
    
    @NotNull(message = "Event start time is required")
    @Column(name = "start_time", nullable = false)
    private LocalDateTime startTime;
    
    @Column(name = "end_time")
    private LocalDateTime endTime;
    
    @Size(max = 500, message = "Location cannot exceed 500 characters")
    @Column(name = "location", length = 500)
    private String location;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "creator_id", nullable = false, referencedColumnName = "id")
    private User creator;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "group_id", referencedColumnName = "id")
    private ChatGroup group;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private EventStatus status = EventStatus.SCHEDULED;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "category", nullable = false)
    private EventCategory category = EventCategory.GENERAL;
    
    @Column(name = "max_attendees")
    private Integer maxAttendees;
    
    @Column(name = "is_recurring", nullable = false)
    private Boolean isRecurring = false;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "recurrence_type")
    private RecurrenceType recurrenceType;
    
    @Column(name = "recurrence_end_date")
    private LocalDateTime recurrenceEndDate;
    
    @Column(name = "requires_approval", nullable = false)
    private Boolean requiresApproval = false;
    
    @Column(name = "bring_list_enabled", nullable = false)
    private Boolean bringListEnabled = false;
    
    @OneToMany(mappedBy = "event", cascade = CascadeType.ALL, orphanRemoval = true)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private List<EventBringItem> bringItems = new ArrayList<>();
    
    @Column(name = "original_category")
    private String originalCategory;
    
    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    public enum EventStatus {
        SCHEDULED,
        IN_PROGRESS,
        COMPLETED,
        CANCELLED,
        POSTPONED
    }
    
    public enum EventCategory {
        GENERAL,
        WORSHIP,
        BIBLE_STUDY,
        PRAYER,
        FELLOWSHIP,
        OUTREACH,
        YOUTH,
        CHILDREN,
        MENS,
        WOMENS,
        SENIORS,
        MISSIONS,
        MINISTRY,
        SOCIAL,
        EDUCATION,
        MUSIC,
        OTHER,
        // Legacy values for backward compatibility
        MENS_MINISTRY,
        WOMENS_MINISTRY,
        SPECIAL_EVENT,
        MEETING,
        VOLUNTEER
    }
    
    public enum RecurrenceType {
        DAILY,
        WEEKLY,
        MONTHLY,
        YEARLY
    }
}